Panic in logging plugin

```
2023-02-15T07:16:35Z INFO plugin.logging gateway/plugin.go:218 logging plugin enabled
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x38 pc=0x1edda9a]

goroutine 1 [running]:
github.com/rancher/opni/plugins/logging/pkg/gateway.Scheme({0x28f6ad8, 0xc000692550})
	github.com/rancher/opni/plugins/logging/pkg/gateway/plugin.go:237 +0x33a
github.com/rancher/opni/pkg/plugins.(*Main).Exec(0xc000723f60)
	github.com/rancher/opni/pkg/plugins/core.go:80 +0x2bf
main.main()
	github.com/rancher/opni/plugins/logging/main.go:17 +0xbf
2023-02-15T07:16:35.227Z [ERROR] plugin process exited: path=/var/lib/opni/plugins/plugin_logging pid=77 error="exit status 2"
2023-02-15T07:16:35Z ERROR gateway.pluginloader plugins/loader.go:157 failed to load plugin {"plugin": "github.com/rancher/opni/plugins/logging", "error": "Unrecognized remote plugin message: \nThis usually means\n  the plugin was not compiled for this architecture,\n  the plugin is missing dynamic-link libraries necessary to run,\n  the plugin is not executable by this process due to file permissions, or\n  the plugin failed to negotiate the initial go-plugin protocol handshake\n\nAdditional notes about plugin:\n  Path: /var/lib/opni/plugins/plugin_logging\n  Mode: -rwxr-xr-x\n  Owner: 0 [root] (current: 0 [root])\n  Group: 0 [root] (current: 0 [root])\n  ELF architecture: EM_X86_64 (current architecture: amd64)\n"}
```

--- a/plugins/logging/pkg/agent/plugin.go
+++ b/plugins/logging/pkg/agent/plugin.go
@@ -13,6 +13,7 @@ import (
 	"github.com/rancher/opni/plugins/logging/pkg/agent/drivers/events"
 	"github.com/rancher/opni/plugins/logging/pkg/agent/drivers/kubernetes"
 	loggingutil "github.com/rancher/opni/plugins/logging/pkg/util"
+	collogspb "go.opentelemetry.io/proto/otlp/collector/logs/v1"
 	"go.uber.org/zap"
 )
 
@@ -65,6 +66,8 @@ func NewPlugin(ctx context.Context) *Plugin {
 
 	return p
 }
+
+var _ collogspb.LogsServiceServer = (*loggingutil.OTELForwarder)(nil)
 
 func Scheme(ctx context.Context) meta.Scheme {
 	scheme := meta.NewScheme(meta.WithMode(meta.ModeAgent))

--- a/plugins/logging/pkg/gateway/drivers/kubernetes_manager.go
+++ b/plugins/logging/pkg/gateway/drivers/kubernetes_manager.go
@@ -37,7 +37,7 @@ type KubernetesManagerDriverOptions struct {
 	k8sClient         client.Client
 	namespace         string
 	opensearchCluster *opnimeta.OpensearchClusterRef
-	logger            zap.SugaredLogger
+	logger            *zap.SugaredLogger
 }
 
 type KubernetesManagerDriverOption func(*KubernetesManagerDriverOptions)
@@ -65,7 +65,7 @@ func WithOpensearchCluster(opensearchCluster *opnimeta.OpensearchClusterRef) Kub
 		o.opensearchCluster = opensearchCluster
 	}
 }
-func WithLogger(logger zap.SugaredLogger) KubernetesManagerDriverOption {
+func WithLogger(logger *zap.SugaredLogger) KubernetesManagerDriverOption {
 	return func(o *KubernetesManagerDriverOptions) {
 		o.logger = logger
 	}

--- a/plugins/logging/pkg/gateway/plugin.go
+++ b/plugins/logging/pkg/gateway/plugin.go
@@ -59,6 +59,7 @@ type Plugin struct {
 	opensearchManager   *opensearchdata.Manager
 	logging             backend.LoggingBackend
 	otelForwarder       *loggingutil.OTELForwarder
+	clusterDriver       drivers.ClusterDriver
 }
 
 type PluginOptions struct {
@@ -151,6 +152,12 @@ func NewPlugin(ctx context.Context, opts ...PluginOption) *Plugin {
 		os.Exit(1)
 	}
 
+	clusterDriver, _ := drivers.NewKubernetesManagerDriver(
+		drivers.WithK8sClient(cli),
+		drivers.WithLogger(lg.Named("cluster-driver")),
+		drivers.WithOpensearchCluster(options.opensearchCluster),
+	)
+
 	p := &Plugin{
 		PluginOptions:       options,
 		ctx:                 ctx,
@@ -168,6 +175,7 @@ func NewPlugin(ctx context.Context, opts ...PluginOption) *Plugin {
 			loggingutil.WithLogger(lg.Named("otel-forwarder")),
 			loggingutil.WithAddress(fmt.Sprintf("http://%s:%d", OpniPreprocessingAddress, OpniPreprocessingPort)),
 		),
+		clusterDriver: clusterDriver,
 	}
 
 	future.Wait4(p.storageBackend, p.mgmtApi, p.uninstallController, p.nodeManagerClient,
@@ -177,11 +185,6 @@ func NewPlugin(ctx context.Context, opts ...PluginOption) *Plugin {
 			uninstallController *task.Controller,
 			nodeManagerClient capabilityv1.NodeManagerClient,
 		) {
-			clusterDriver, _ := drivers.NewKubernetesManagerDriver(
-				drivers.WithK8sClient(p.k8sClient),
-				drivers.WithLogger(*p.logger.Named("cluster-driver")),
-				drivers.WithOpensearchCluster(p.opensearchCluster),
-			)
 			p.logging.Initialize(backend.LoggingBackendConfig{
 				Logger:              p.logger.Named("logging-backend"),
 				StorageBackend:      storageBackend,
@@ -189,7 +192,7 @@ func NewPlugin(ctx context.Context, opts ...PluginOption) *Plugin {
 				OpensearchCluster:   p.opensearchCluster,
 				MgmtClient:          mgmtClient,
 				NodeManagerClient:   nodeManagerClient,
-				ClusterDriver:       clusterDriver,
+				ClusterDriver:       p.clusterDriver,
 			})
 		},
 	)
@@ -234,7 +237,7 @@ func Scheme(ctx context.Context) meta.Scheme {
 
 	loggingManager := p.NewLoggingManagerForPlugin()
 
-	if state := p.logging.ClusterDriver.GetInstallStatus(ctx); state == drivers.Installed {
+	if state := p.clusterDriver.GetInstallStatus(ctx); state == drivers.Installed {
 		go p.opensearchManager.SetClient(loggingManager.setOpensearchClient)
 		if p.opensearchManager.ShouldCreateInitialAdmin() {
 			err = loggingManager.createInitialAdmin()

--- a/plugins/logging/pkg/gateway/plugin.go
+++ b/plugins/logging/pkg/gateway/plugin.go
@@ -34,6 +34,7 @@ import (
 	"github.com/rancher/opni/plugins/logging/pkg/gateway/drivers"
 	"github.com/rancher/opni/plugins/logging/pkg/opensearchdata"
 	loggingutil "github.com/rancher/opni/plugins/logging/pkg/util"
+	collogspb "go.opentelemetry.io/proto/otlp/collector/logs/v1"
 	corev1 "k8s.io/api/core/v1"
 )
 
@@ -202,6 +203,7 @@ func NewPlugin(ctx context.Context, opts ...PluginOption) *Plugin {
 
 var _ loggingadmin.LoggingAdminServer = (*Plugin)(nil)
 var _ loggingadmin.LoggingAdminV2Server = (*LoggingManagerV2)(nil)
+var _ collogspb.LogsServiceServer = (*loggingutil.OTELForwarder)(nil)
 
 func Scheme(ctx context.Context) meta.Scheme {
 	scheme := meta.NewScheme(meta.WithMode(meta.ModeGateway))

--- a/plugins/logging/pkg/util/otel.go
+++ b/plugins/logging/pkg/util/otel.go
@@ -17,6 +17,7 @@ const (
 )
 
 type OTELForwarder struct {
+	collogspb.UnsafeLogsServiceServer
 	opts otelForwarderOptions
 
 	Client *AsyncClient[collogspb.LogsServiceClient]

